protocols/identify/src/handler.rs

```diff
@@ -152,14 +177,23 @@ impl Handler {
         >,
     ) {
         match output {
-            future::Either::Left(substream) => {
+            future::Either::Left(stream) => {
                 let info = self.build_info();
 
-                self.pending_replies
-                    .push(crate::protocol::send(substream, info).boxed());
+                if self
+                    .pending_replies
+                    .try_push(crate::protocol::send(stream, info))
+                    .is_err()
+                {
+                    warn!("Dropping inbound stream because we are at capacity");
+                }
             }
-            future::Either::Right(fut) => {
-                if self.inbound_identify_push.replace(fut).is_some() {
+            future::Either::Right(stream) => {
+                if self
+                    .inbound_identify_push
+                    .replace(recv_push(stream).boxed())
+                    .is_some()
```

Comment:
In https://github.com/libp2p/rust-libp2p/pull/2694, we introduced at most a single identify push running at a time. Just recently, we introduced support for partial pushes in https://github.com/libp2p/rust-libp2p/pull/4495 which counters the reasoning of #2694.

I think this warrants revisiting this here and moving the inbound pushes also to a `FuturesSet`.

Thoughts @mxinden?

Reply:
Yes, this is a good thought!
I think Thomas is right. After #4495 replacing pushes is just a mistake because every one of them can contain different info.

Reply:
Fine by me to introduce more than one concurrent identify push, e.g. by switching to `FuturesSet`. Only requirement is that the `FuturesSet` has a reasonable bound, e.g. `10`. Given that `libp2p-identify` does a whole sync periodically, I don't think we need to worry about inconsistencies nor race conditions between two concurrent pushes.